worker/worker.go

```diff
@@ -425,18 +454,32 @@ func (w *worker) rhpRenewHandler(jc jape.Context) {
 	if jc.Decode(&rrr) != nil {
 		return
 	}
-	var cs consensus.State
-	cs.Index.Height = uint64(rrr.TransactionSet[len(rrr.TransactionSet)-1].FileContracts[0].WindowStart)
+	hostSettings, hostKey, rk, toRenewID, renterFunds := rrr.HostSettings, rrr.HostKey, rrr.RenterKey, rrr.ContractID, rrr.RenterFunds
+	renterAddress, endHeight := rrr.RenterAddress, rrr.EndHeight
 
 	var contract rhpv2.ContractRevision
 	var txnSet []types.Transaction
-	err := w.withTransportV2(jc.Request.Context(), rrr.HostIP, rrr.HostKey, func(t *rhpv2.Transport) error {
-		session, err := rhpv2.RPCLock(jc.Request.Context(), t, rrr.ContractID, rrr.RenterKey, 5*time.Second)
+	err := w.withTransportV2(jc.Request.Context(), hostSettings.NetAddress, hostKey, func(t *rhpv2.Transport) error {
+		session, err := rhpv2.RPCLock(jc.Request.Context(), t, toRenewID, rk, 5*time.Second)
 		if err != nil {
 			return err
 		}
-		contract, txnSet, err = session.RenewContract(cs, rrr.TransactionSet, rrr.FinalPayment)
-		return err
+		rev := session.Contract()
```

Comment:
seems like `Revision` and `Contract` are interchangeable in the worker, why are we calling a revision a contract, even the `RHPRenewResponse` returns the revision under the `contract` key